components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -375,17 +375,13 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
 
             String idToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
 
-            if (StringUtils.isBlank(idToken) && requiredIDToken(authenticatorProperties)) {
```

Comment:
Why did we remove this check?

Reply:
This is a redundant check? If id_token becomes blank,  context.setProperty(OIDCAuthenticatorConstants.ID_TOKEN, idToken);  will not work due to this first check..

Reply:
But in the previous case, it throws an exception and breaks the flow. With this change it allows to complete the flow even with the empty id_token